### PR TITLE
Add httpStatus to TargomoClientException

### PR DESCRIPTION
**Pull request by GideonTargomo**
Adds httpStatus as a potential variable in `TargomoClientException` to be used in some exception handling. Http codes added in reachability and statistics requests

### CODE REVIEW

File: src/main/java/com/targomo/client/api/exception/TargomoClientException.java [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
 public class TargomoClientException extends Exception {
 
 	private static final long serialVersionUID = 748077356312331362L;
+	
+	@Getter
+	private int httpStatusCode;
 
 	public TargomoClientException(String message, Throwable throwable) {
-		super(message, throwable);
+		this(message, throwable, Constants.BAD_GATEWAY_CODE);
```
Comment (DavidRedlich) [inline src/main/java/com/targomo/client/api/exception/TargomoClientException.java]: would be nice to have a small comment here: Default is that this Exception is tarnslated into a BadGateway response (or something like that)

Comment (GideonTargomo) [reply src/main/java/com/targomo/client/api/exception/TargomoClientException.java]: Updated and added full javadoc

File: src/main/java/com/targomo/client/api/request/ReachabilityRequest.java [hunk 2]
```diff
@@ -116,7 +116,7 @@ private ReachabilityResponse validateResponse(final Response response,
 		} else if (response.getStatus() == Response.Status.GATEWAY_TIMEOUT.getStatusCode()) {
 			return new ReachabilityResponse(travelOptions, "gateway-time-out", roundTripTime, requestStart);
 		} else {
-			throw new TargomoClientException(response.readEntity(String.class), null);
+			throw new TargomoClientException(response.readEntity(String.class), null, response.getStatus());
```
Comment (DavidRedlich) [inline src/main/java/com/targomo/client/api/request/ReachabilityRequest.java]: you now have a constructor where you don't have to set throwable to null

File: src/main/java/com/targomo/client/api/request/StatisticsRequest.java [hunk 3]
```diff
@@ -143,7 +143,7 @@ else if (response.getStatus() == Response.Status.GATEWAY_TIMEOUT.getStatusCode()
 			return new StatisticsResponse(travelOptions, "gateway-time-out", roundTripTime, requestStart);
 		}
 		else {
-			throw new TargomoClientException(response.readEntity(String.class), null);
+			throw new TargomoClientException(response.readEntity(String.class), null, response.getStatus());
```
Comment (DavidRedlich) [inline src/main/java/com/targomo/client/api/request/StatisticsRequest.java]: same here

**Pull request merged**